File: wallet/src/main/java/com/worldventures/wallet/service/nxt/util/TokenizedRecord.java
package com.worldventures.wallet.service.nxt.util;

import android.support.annotation.NonNull;

import com.worldventures.wallet.domain.entity.record.Record;
import com.worldventures.wallet.service.nxt.model.MultiResponseBody;

import java.util.Collections;
import java.util.List;

public final class TokenizedRecord extends NxtRecordResponse {

   public static TokenizedRecord from(@NonNull Record detokenizedRecord, @NonNull MultiResponseBody nxtResponses) {
      return from(detokenizedRecord, Collections.singletonList(nxtResponses));
   }

   public static TokenizedRecord from(@NonNull Record detokenizedRecord, @NonNull List<MultiResponseBody> nxtResponses) {
      return new TokenizedRecord(detokenizedRecord, nxtResponses);
   }

   private TokenizedRecord(@NonNull Record detokenizedRecord, @NonNull List<MultiResponseBody> nxtResponses) {
      super(detokenizedRecord, nxtResponses);
   }

   @Override
   public Record getTokenizedRecord() {
      return NxtBankCardHelper.getTokenizedRecord(this, refIdPrefix);
   }

   @Override
   public Record getDetokenizedRecord() {
      return record;
   }

}

File: wallet/src/main/java/com/worldventures/wallet/service/nxt/util/TokenizedRecord.kt
package com.worldventures.wallet.service.nxt.util

import com.worldventures.wallet.domain.entity.record.Record
import com.worldventures.wallet.service.nxt.model.MultiResponseBody

class TokenizedRecord private constructor(detokenizedRecord: Record, nxtResponses: List<MultiResponseBody>)
   : NxtRecordResponse(detokenizedRecord, nxtResponses) {

   override fun getTokenizedRecord(): Record {
      return NxtBankCardHelper.getTokenizedRecord(this, refIdPrefix)
   }

   override fun getDetokenizedRecord(): Record {
      return record
   }

   companion object {

      fun from(detokenizedRecord: Record, nxtResponses: MultiResponseBody): TokenizedRecord {
         return from(detokenizedRecord, listOf(nxtResponses))
      }

      fun from(detokenizedRecord: Record, nxtResponses: List<MultiResponseBody>): TokenizedRecord {
         return TokenizedRecord(detokenizedRecord, nxtResponses)
      }
   }

}
